### ISSUE TITLE
@defgroup contained in the markdown mainpage.md are not including the @ingroup marked classes in the generated documentation.

### ISSUE THREAD

**Issue by damiandixon** [doxygen/doxygen#8371]
**Describe the bug**
`@defgroup` contained in the markdown `mainpage.md` are not including the `@ingroup` marked classes in the generated documentation.

**Expected behavior**
I would expect the behaviour that I see in versions prior to 1.8.17.

The problem is present in 1.9.1.

**Screenshots**
Incorrect output from 1.8.17:

![image](https://user-images.githubusercontent.com/1237723/106777017-81ff5f80-663c-11eb-8419-0c4e1e47295c.png)

Correct output from 1.8.16:

![image](https://user-images.githubusercontent.com/1237723/106777084-93486c00-663c-11eb-81fd-4916fd0d3017.png)

**To Reproduce**
Sample attached. With output from 1.8.17 and 1.8.16

[ingroup_problem.tar.gz](https://github.com/doxygen/doxygen/files/5919987/ingroup_problem.tar.gz)

**Version**
Running on Ubuntu 20.04.

The problem occurs in 1.8.17 and latest.

The problem does not occur in 1.8.16

The problem appears to have been introduced in d38df57916d5902abc9dbaf5891c14898ce675da.

**Additional context**
I believe that the change in the file doxygen.cpp at line 3651 may not be correct (guessing)...

![image](https://user-images.githubusercontent.com/1237723/106777184-aa875980-663c-11eb-8797-bca296fd6fb2.png)

The removal of: `root->groups->append(new Grouping(md->getGroupDef()->name(), md->getGroupPri()));`

**Comment by albert-github** [comment]
Looks indeed a bit strange that the line 
```
root->groups->append(new Grouping(md->getGroupDef()->name(), md->getGroupPri()));
```
disappeared (now around line 3534)

**Comment by doxygen** [comment]
@damiandixon Thanks for the detailed issue description. Please verify if the provided fix indeed solves the problem for you. Please keep the issue open; it will be closed automatically when the next official release becomes available.

**Comment by damiandixon** [comment]
@doxygen Looks like that fixes the problem with both the test case and with my larger projects.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
